server/src/test/java/org/elasticsearch/bootstrap/BootstrapChecksTests.java

```diff
@@ -194,6 +215,7 @@ boolean isMemoryLocked() {
                 "initial heap size [" + initialHeapSize.get() + "] " + "not equal to maximum heap size [" + maxHeapSize.get() + "]"
             )
         );
+        assertThat(e.getMessage(), containsString("; for more information see [" + check.referenceDocs() + "]"));
```

Comment:
Suggest making each of these assert that the message contains a genuine link (i.e. at least it's not `null`) rather than just copying `check.referenceDocs()`:

```suggestion
        assertThat(e.getMessage(), containsString("; for more information see [https://www.elastic.co/guide/en/elasticsearch/reference/"));
```

Reply:
Thanks for your suggestion. It might be better to make each of these assert that the message contains a full link.
```suggestion
        assertThat(e.getMessage(), containsString("; for more information see [https://www.elastic.co/guide/en/elasticsearch/reference/master/_heap_size_check.html]"));
```

Reply:
That would be a little fragile for my taste: if we moved these docs around in future then we'd have to update the link in `reference-docs-links.json` _and_ in these tests, but that seems unnecessary because we're already validating the links elsewhere in the build. Just checking the prefix ensures that it's (a) not null and (b) mentioned in the message and that's enough IMO.